CMakeLists.txt

```diff
@@ -309,29 +309,24 @@ add_custom_command(OUTPUT ${stamp_file}
                    COMMAND ${CMAKE_COMMAND} -E copy_directory ${CMAKE_SOURCE_DIR}/README ${CMAKE_BINARY_DIR}/README
                    COMMAND ${CMAKE_COMMAND} -E touch ${stamp_file}
                    COMMENT "Copying directories such as etc, icons, fonts, js, ui5, etc. to build area")
+add_custom_target(move_artifacts DEPENDS ${stamp_file})
 
+
+#---Copy files to the build area at cmake configure time---------------------------------
+set(DIRECTORIES_TO_COPY tutorials etc test icons fonts macros)
 if(http)
-set(jsroot_files js/*)
+  list(APPEND DIRECTORIES_TO_COPY js)
 endif()
 if(webgui)
-set(openui5_files ui5/*)
+  list(APPEND DIRECTORIES_TO_COPY ui5)
 endif()
 
-#---Copy files to the build area, with dependency---------------------------------
-file(GLOB_RECURSE artifact_files RELATIVE ${CMAKE_SOURCE_DIR} tutorials/* etc/* test/* icons/* fonts/* macros/* ${jsroot_files} ${openui5_files})
-set(artifact_files_builddir)
-foreach(artifact_file ${artifact_files})
+foreach(DIR_TO_COPY ${DIRECTORIES_TO_COPY})
   # Filter out hsimple.root; someone might have created it in the src dir, and the hsimple.root
   # target below will interfere.
-  if (NOT (artifact_file STREQUAL "tutorials/hsimple.root"))
-    add_custom_command(OUTPUT ${CMAKE_BINARY_DIR}/${artifact_file}
-      COMMAND ${CMAKE_COMMAND} -E copy_if_different ${CMAKE_SOURCE_DIR}/${artifact_file} ${CMAKE_BINARY_DIR}/${artifact_file}
-      COMMENT "Copying ${CMAKE_SOURCE_DIR}/${artifact_file}"
-      DEPENDS ${CMAKE_SOURCE_DIR}/${artifact_file})
-    list(APPEND artifact_files_builddir ${CMAKE_BINARY_DIR}/${artifact_file})
-  endif()
+  file(COPY ${DIR_TO_COPY} DESTINATION ${CMAKE_BINARY_DIR}
```

Comment:
The `file(COPY_FILE` command as a `ONLY_IF_DIFFERENT` options, does this syntax  means that the file are copied at reconfigure time?

Reply:
I guess the question is whether they should be recopied every time. Currently that's the case, so one could move to `COPY_FILE ... ONLY_IF_DIFFERENT`.
However, this is not supported in CMake 3.20, which is our minimum CMake version.

Reply:
This does not apply, any more, since this part will be factored out.